prow/logrusutil/logrusutil.go

```diff
@@ -73,3 +77,41 @@ func (f *DefaultFieldsFormatter) Format(entry *logrus.Entry) ([]byte, error) {
 		Caller:  entry.Caller,
 	})
 }
+
+// CensoringFormatter represents a logrus formatter that
+// can be used to censor sensitive information
+type CensoringFormatter struct {
+	delegate   logrus.Formatter
+	getSecrets func() sets.String
+}
+
+func (f CensoringFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	raw, err := f.delegate.Format(entry)
+	if err != nil {
+		return raw, err
+	}
+	return f.censor(raw), nil
+}
+
+const censored = "CENSORED"
+
+var censoredBytes = []byte(censored)
+
+// Censor replaces sensitive parts of the content with a placeholder.
+func (f CensoringFormatter) censor(content []byte) []byte {
+	if f.getSecrets != nil {
```

Comment:
`f.getSecrets` being `nil` should be an error -- that seems like a programmer error

Reply:
Removed checking.